### DISCUSSION TITLE
Unable to bypass cache checks on certain API routes

### DISCUSSION THREAD

**Discussion by ckohen** [discordjs/discord.js#5242] [General]
**Please describe the problem you are having in as much detail as possible:**
Certain API routes only need the id of a guild / channel / role / user however, the some of the checks use resolve instead of resolveID so raw Snowflakes cannot be passed if the object is not cached

**Include a reproducible code sample here, if possible:**

This is only one example:
```js
channel.createOverwrite('800910187014651999', { 'VIEW_CHANNEL': false } ) // Where the id is not of a cached user
```

**Further details:**

- discord.js version: master (c4c81711)
- Node.js version: v14.15.3
- Operating system: Multiple
- Priority this issue should have – please be realistic and elaborate if possible:
medium - this affects multiple routes listed below and makes it impossible to use uncached id's on these routes

<details>
<summary>Currently Affects the following routes:</summary>

- GuildEmojiManager#create() -  *roles are always expected to be cached*
- GuildEmojiRoleManager#add() -  *roles are always expected to be cached*
- GuildMemberRoleManager#add() -  *roles are always expected to be cached*
- GuildMemberRoleManager#remove() -  *roles are always expected to be cached*
- Guild#addMember() - *roles are always expected to be cached*
- GuildChannel#updateOverwrite() -  *roles are always expected to be cached, this also affects users*
- GuildChannel#createOverwrite() -  *roles are always expected to be cached, this also affects users*
- GuildMember#edit() - *channels are always expected to be cached*

</details>

Since roles are expected to be cached, this is not currently a problem for most of the mentioned methods.
However, with the v13, intents will make it possible (although unlikely) for these roles to be uncached.
Whenever these methods are being used, the appropriate `Resolvable` types accept snowflakes with no indication that the associated object must be cached, likely because they shouldn't need to be

**Possible Fixes**

- Replace the last `resolve` attempt in each chain with `resolveID`, passing the validity check off to discord
- If anyone would like to chime in on why `resolve` is used here (other than attempting type checking) that would help clear up the first option
- Directly check if the passed option is of type Snowflake after attempting to resolve, will require #5157
- Leave as is and note that the associated object must be cached on these methods

**Relevant client options:**

- partials: Can Vary
- gateway intents: Can Vary
- other: none

- [x] I have also tested the issue on latest master, commit hash: c4c81711

**Comment by iShibi** [comment]
#5495 fixed most of the above issues. Once #5489 is merged, `GuildChannel#updateOverwrite()` & `GuildChannel#createOverwrite()` will get fixed too.

**Reply by iShibi** [reply]
Update: Both of the PRs has been merged into master now 🎉
